lib/rules/no-unneeded-ternary.js

```diff
@@ -60,9 +78,21 @@ module.exports = {
 
             ConditionalExpression(node) {
                 if (isBooleanLiteral(node.alternate) && isBooleanLiteral(node.consequent)) {
-                    context.report(node, node.consequent.loc.start, "Unnecessary use of boolean literals in conditional expression.");
+                    context.report({
```

Comment:
I have my doubts about the if check for this report. Can you please check that the fixer behaves correctly for `foo ? false : true`? (And for `foo ? true : true` maybe, but that's a different problem...